modules/charon-core/src/test/java/org/wso2/charon3/core/schema/ServerSideValidatorTest.java

```diff
@@ -0,0 +1,663 @@
+import org.wso2.charon3.core.attributes.ComplexAttribute;
+import org.wso2.charon3.core.attributes.MultiValuedAttribute;
+import org.wso2.charon3.core.attributes.SimpleAttribute;
+import org.wso2.charon3.core.exceptions.BadRequestException;
+import org.wso2.charon3.core.exceptions.CharonException;
+import org.wso2.charon3.core.exceptions.NotFoundException;
+import org.wso2.charon3.core.objects.AbstractSCIMObject;
+import org.wso2.charon3.core.objects.Group;
+import org.wso2.charon3.core.objects.Role;
+import org.wso2.charon3.core.objects.User;
+import org.wso2.charon3.core.protocol.endpoints.AbstractResourceManager;
+import org.wso2.charon3.core.utils.CopyUtil;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.DataType.COMPLEX;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.DataType.REFERENCE;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.DataType.STRING;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Mutability.IMMUTABLE;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Mutability.READ_ONLY;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Mutability.READ_WRITE;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Returned.ALWAYS;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Returned.DEFAULT;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Uniqueness.NONE;
+import static org.wso2.charon3.core.schema.SCIMDefinitions.Uniqueness.SERVER;
+
+/**
+ * Test class of ServerSideValidator.
+ */
+@PrepareForTest({AbstractResourceManager.class})
+public class ServerSideValidatorTest extends PowerMockTestCase {
+
+    @DataProvider(name = "dataForValidateCreatedSCIMObjectSuccess")
+    public Object[][] dataToValidateCreatedSCIMObjectSuccess() throws InstantiationException, IllegalAccessException {
+
+        return new Object[][]{
+
+                {createNewUser(), createSCIMResourceTypeSchemaUser()},
+                {createNewGroup(), createSCIMResourceTypeSchemaGroup()},
+                {createNewRole(), createSCIMResourceTypeSchemaRole()}
+        };
+    }
+
+    @Test(dataProvider = "dataForValidateCreatedSCIMObjectSuccess")
+    public void testValidateCreatedSCIMObjectSuccess(Object objectScimObject, Object objectResourceSchema)
+            throws CharonException, BadRequestException, NotFoundException {
+
+        AbstractSCIMObject scimObject = (AbstractSCIMObject) objectScimObject;
+        SCIMResourceTypeSchema resourceSchema = (SCIMResourceTypeSchema) objectResourceSchema;
+
+        mockStatic(AbstractResourceManager.class);
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT)).thenReturn(
+                "https://localhost:9443/scim2/Users");
```

Comment:
This and the following line breakings are not in the expected format. Please recheck. Check other places as well.
```suggestion
        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
                .thenReturn("https://localhost:9443/scim2/Users");
```

Reply:
fixed in abc18d12f649b0a2709ddcde8561827e72d4cc36